utility/src/main/java/org/oppia/android/util/parser/html/HtmlParser.kt

```diff
@@ -134,7 +135,8 @@ class HtmlParser private constructor(
         consoleLogger,
         context.assets,
         htmlContentTextView.lineHeight.toFloat(),
-        cacheLatexRendering
+        cacheLatexRendering,
+        context as? Application ?: context.applicationContext as Application
```

Comment:
Can you expand on why this is needed? I'd expect context to be the application context in this case based on how we set up the Dagger graph, but perhaps I'm missing something.

Reply:
Explain in above comment, as we need context so there is way to pass the context.

Reply:
Sorry, I'm not sure if the other explanation actually answers my question. Why can't ``context`` be used instead of the extra casting work to ``Application`` that's happening here?

Reply:
Actually here we need the `Application Context` to access the `application.resources` in `ResourcesCompat.getColor` That's why I using `Application` as with only `Context` these resources is not accessible.